### PR TITLE
Dask-MPI PR

### PR DESCRIPTION
**Pull request by kmpaul**

### CODE REVIEW

**Comment by mrocklin** [comment]
From this error it appears that we can't use noarch if we have an executable?

> [noarch_python] Noarch package contains binary script: dask-mpi.exe

Or at least that's how I interpret that line.  It might be worth removing the noarch line from the yaml file.

File: recipes/dask-mpi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 77afcf670017c2ef2139732091b743de96ff4a122fe7156275c84b4df745389c
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (jakirkham) [inline recipes/dask-mpi/meta.yaml]: Would add the following...

```yaml
  entry_points:
    - dask-mpi = dask_mpi.cli:go
```

Comment (jakirkham) [reply recipes/dask-mpi/meta.yaml]: Would add this change should allow `noarch: python` to be added back.

File: recipes/dask-mpi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - six
+  run:
+    - python
+    - dask
+    - distributed
+    - mpi4py
+
+test:
+  imports:
+    - dask_mpi
```
Comment (jakirkham) [inline recipes/dask-mpi/meta.yaml]: Would add a test for the CLI like so.

```yaml
  commands:
    - dask-mpi --help
```

**Comment by jakirkham** [comment]
Looks pretty good to me. Couple minor comments above. Please let me know if you have any questions :)

File: recipes/dask-mpi/meta.yaml (hunk 1 above)
Comment (kmpaul) [reply recipes/dask-mpi/meta.yaml]: Thanks, done!

File: recipes/dask-mpi/meta.yaml (hunk 2 above)
Comment (kmpaul) [reply recipes/dask-mpi/meta.yaml]: Also done.  Thanks!

**Comment by mrocklin** [comment]
On Windows

> conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {'mpich'}

I take it that we *do* need to exclude Windows somewhere in the recipe then?

**Comment by kmpaul** [comment]
Yes.  It looks like it.  If we want to deal with mpi4py, I think we need to limit it to not windows.  (And I'll have to get back to this on Monday.)

File: recipes/dask-mpi/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 77afcf670017c2ef2139732091b743de96ff4a122fe7156275c84b4df745389c
+
+build:
```
[the remark this answers is not in the document]
Comment (kmpaul) [reply recipes/dask-mpi/meta.yaml]: Thanks, @willirath!  Trying again...

File: recipes/dask-mpi/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  skip: true  # [win]
+  noarch: python
```
Comment (mrocklin) [inline recipes/dask-mpi/meta.yaml]: We need to remove this line.

```suggestion
```

**Comment by mrocklin** [comment]
Tests pass!

**Comment by jakirkham** [comment]
We decided a while back that it is not a blocker to make a package `noarch` even if some of its dependencies cannot be satisfied on all platforms. That said, can understand if you would feel more comfortable making it an arch specific package instead.

**Comment by jakirkham** [comment]
Should add would be happy to merge as is or merge with `noarch` added back (even though Windows will fail to install it). Please let me know what you decide.

**Comment by mrocklin** [comment]
Happy either way.  I recommend that we merge this now and add noarch in the
futue.

**Comment by jakirkham** [comment]
Thanks all :)

**Pull request merged**